Xamarin.PropertyEditing.Mac/Controls/Custom/NumericTextField.cs

```diff
@@ -54,32 +54,78 @@ public override bool ShouldBeginEditing (NSText fieldEditor)
 
 		protected virtual void OnKeyArrowUp ()
 		{
-			var handler = KeyArrowUp;
-			if (handler != null) {
-				handler (this, EventArgs.Empty);
-			}
+			KeyArrowUp?.Invoke (this, EventArgs.Empty);
 		}
 
 		protected virtual void OnKeyArrowDown ()
 		{
-			var handler = KeyArrowDown;
-			if (handler != null) {
-				handler (this, EventArgs.Empty);
-			}
+			KeyArrowDown?.Invoke (this, EventArgs.Empty);
 		}
 
 		public virtual void RaiseValidatedEditingEnded ()
 		{
-			var handler = ValidatedEditingEnded;
-			if (handler != null)
-				handler (this, EventArgs.Empty);
+			ValidatedEditingEnded?.Invoke (this, EventArgs.Empty);
 		}
 
 		public virtual void ResetInvalidInput ()
 		{
 			this.StringValue = cachedValueString;
 		}
 
+		public static bool CheckIfNumber (string finalString, ValidationType mode, bool allowNegativeValues)
+		{
+			return mode == ValidationType.Decimal ?
+				ValidateDecimal (finalString, allowNegativeValues) :
+				ValidateInteger (finalString, allowNegativeValues);
+		}
+
+		public static bool ValidateDecimal (string finalString, bool allowNegativeValues)
+		{
+			double value;
+			//Checks parsing to number
+			if (!double.TryParse (finalString, System.Globalization.NumberStyles.Any, System.Globalization.CultureInfo.CurrentUICulture, out value))
+				return false;
+			//Checks if needs to be possitive value
+			if (!allowNegativeValues && value < 0)
+				return false;
+			//Checks a common validation
+			return ViewModels.RatioViewModel.IsValidInput (finalString);
+		}
+
+		public static bool ValidateInteger (string finalString, bool allowNegativeValues)
+		{
+			int value;
+			//Checks parsing to number
+			if (!int.TryParse (finalString, out value))
+				return false;
+			//Checks if needs to be possitive value
+			if (!allowNegativeValues && value < 0)
+				return false;
+			//Checks a common validation
+			return ViewModels.RatioViewModel.IsValidInput (finalString);
+		}
+
+		public static bool CheckIfRatio (string finalString, ValidationType mode, bool allowNegativeValues)
+		{
+			var parts = finalString.Split (ViewModels.RatioViewModel.Separators);
```

Comment:
I think the handling of spaces got lost in the new implementation. You may need a second char array that also has spaces. Also, I think you should use the overload that tells `Split` to remove empty entries (`StringSplitOptions. RemoveEmptyEntries`).

Reply:
This is a (hopefully quick) high level check, done at the control level. The full check is done in the VM here -  https://github.com/xamarin/Xamarin.PropertyEditing/pull/74/files/9257d48169ec83a77d98286fd094fa5c73aa5ee7#diff-995a4c01ec15e058d67645b539295624R102

Reply:
Ah, OK, I did miss that, but in fact, I think you can avoid the `Replace` altogether by including space in the split chars.

Reply:
I did think about adding a 2nd array to include the space, but I thought the code would be simpler to maintain without needing 2 arrays, or if having one 3 element array we'd need to generate a temporary 2 element array when checking for valid separators. 
Is your concern the speed overhead of the `Replace` call?

Reply:
No, I don't think speed is a concern on this code path. I'm mostly nitpicking at this point. It's your call really, but having the second array right next to the first one didn't seem necessarily like it would cause any big maintenance problems.

Reply:
OK, You've convinced me :)